### ISSUE TITLE
Dark/light detection doesn't work when referring to colours by name

### ISSUE THREAD

**Issue by amagee** [grommet/grommet#4997]
Grommet appears to detect a button's background color in order to determine whether to use a light or dark font color on it. This works as expected if I use a hex code to define the color but not if I use a named color like "purple".

### Expected Behavior

If the background color is defined as "purple", the text color should be light.

### Actual Behavior

The text color is dark. But if I change the color from "purple" to "#800080" (the same color), the text colour becomes light.

### URL, screen shot, or Codepen exhibiting the issue

https://codesandbox.io/s/grommet-v2-template-forked-5uh9q?file=/index.js

### Your Environment

- Grommet version: 2.6.2, 2.16.3
- Browser Name and version: Chrome 88.0.4324.182
- Operating System and version: Ubuntu 20.10

**Comment by ShimiSun** [comment]
Interesting feedback!! Thinking about this issue, here are my thoughts:
Our color detection algorithm is adaptable to HEX colors and the defined colors of the theme, the color that is being used in your code example is not defined on your theme colors, so grommet assumes you would just like to fall back to the browser's default color of `purple`, and there is no way for grommet to detect what color would that be on the browser so it can't adjust the text color to it.

One suggestion,
If you define what `purple` is, grommet would know to detect its actual color and the detection by name would work as expected so replace this code:
```

**Comment by amagee** [comment]
> there is no way for grommet to detect what color would that be on the browser 

If I'm not mistaken the named browser colours are standard across browsers (eg. "purple" is always `#800080`) so it would be possible for grommet to do the transform and have things just work. The npm package `color-name` appears to be a good option for doing this - it's basically just a big mapping of color names to hex codes ([see here](https://github.com/colorjs/color-name/blob/master/index.js)).

Would this be something you'd be interested in including?

**Comment by ShimiSun** [comment]
Our overall inclination is not to add more packages to our dependencies, so we can keep our package tidy. 
 We will take a look into it though.

**Comment by amagee** [comment]
Yeah, that's understandable. For what it's worth, that `color-name` package is only a couple of kb and it has no dependencies of its own. I feel like supporting named colours would save a bunch of confusion (it took me a while to figure out *why* the text colour wasn't behaving as expected). So while it's definitely not a must-have feature as such, having it would avoid that experience.

**Comment by ericsoderberghp** [comment]
I tend to agree with @ShimiSun on this. There are a lot of small packages we could add for non-80% use cases that we avoid to keep the size of our dependencies down. We are more oriented towards creating a theme for a particular brand or style and naming the colors within that, where the HEX values are, rather than just using the web color palette names.

**Comment by amagee** [comment]
If you really don't want to support these colours I would suggest at least having some kind of warning to save people from having to do a bunch of trial & error debugging to figure out *why* their colours aren't working, especially since people are most likely to try using named colours when they're doing some initial fiddling around with the framework and are least familiar with how things work.

**Comment by ericsoderberghp** [comment]
That's a reasonable suggestion. A scenario I've seen before is where there's a typo between the color name in the theme and it being used in the code.

**Comment by ShimiSun** [comment]
If the user is using the browser color intentionally, I don't think it will be fair to spam their console with warnings. what am I missing?

**Comment by ericsoderberghp** [comment]
Hmm... good point.

**Comment by amagee** [comment]
I doubt anyone is going to want dark text on a purple background intentionally.

The issue is that the colours do not work properly, and additionally it is impossible for someone to discover why their colours are not working properly without doing a bunch of trial & error debugging and guessing.

I still think ideally they would just work properly, but failing that, informing the user why their colours are not working as they expect would be a substantial improvement.
